### PR TITLE
[KGP-538] Approve banner

### PR DESCRIPTION
**Pull request by nguyenhuudungz**

### CODE REVIEW

File: src/hooks/useSwapV2Callback.ts [hunk 1]
```diff
@@ -14,6 +14,8 @@ import { Aggregator } from 'utils/aggregator'
 import { formatCurrencyAmount } from 'utils/formatCurrencyAmount'
 import { sendEVMTransaction, sendSolanaTransactions } from 'utils/sendTransaction'
 
+import { NETWORKS_INFO } from '../constants/networks'
+import { EVMNetworkInfo } from '../constants/networks/type'
```
Comment (namgold) [inline src/hooks/useSwapV2Callback.ts:18]: should use absolute

File: src/hooks/useApproveCallback.ts [hunk 2]
```diff
@@ -15,6 +15,7 @@ import { calculateGasMargin } from 'utils'
 import { Aggregator } from 'utils/aggregator'
 import { computeSlippageAdjustedAmounts } from 'utils/prices'
 
+import { NETWORKS_INFO } from '../constants/networks'
```
Comment (namgold) [inline src/hooks/useApproveCallback.ts:18]:
```suggestion
import { NETWORKS_INFO } from 'constants/networks'
```

File: src/hooks/useApproveCallback.ts [hunk 3]
```diff
@@ -144,12 +145,15 @@ export function useApproveCallbackFromTrade(
 
 // wraps useApproveCallback in the context of a swap
 export function useApproveCallbackFromTradeV2(
+  chainId: ChainId,
```
Comment (namgold) [inline src/hooks/useApproveCallback.ts:148]: get from `useActiveWeb3React`?

File: src/pages/SwapV2/index.tsx [hunk 4]
```diff
@@ -947,6 +958,41 @@ export default function Swap() {
                         </GreyCard>
                       ) : showApproveFlow ? (
                         <>
+                          <Card mb="24px" backgroundColor={rgba(theme.subText, 0.2)} padding="12px">
```
Comment (nguyenhoaidanh) [inline src/pages/SwapV2/index.tsx:961]: move it into a new component to prevent conflict vs HungDoan's pr

Comment (nguyenhuudungz) [reply src/pages/SwapV2/index.tsx]: Done.

**Review by nguyenhoaidanh — APPROVED** [review]

File: src/pages/SwapV2/index.tsx [hunk 5]
```diff
@@ -109,11 +109,14 @@ import { halfAmountSpend, maxAmountSpend } from 'utils/maxAmountSpend'
 import { getSymbolSlug } from 'utils/string'
 import { checkPairInWhiteList } from 'utils/tokenInfo'
 
+import ApproveMessage from '../../components/ApproveMessage'
```
Comment (namgold) [inline src/pages/SwapV2/index.tsx:112]:
```suggestion
import ApproveMessage from 'components/ApproveMessage'
```

File: src/hooks/useSwapV2Callback.ts [hunk 6]
```diff
@@ -118,10 +120,11 @@ export function useSwapV2Callback(
 
     const value = BigNumber.from(trade.inputAmount.currency.isNative ? trade.inputAmount.quotient.toString() : 0)
     const onSwapWithBackendEncode = async (): Promise<string> => {
+      const routerAddress = (NETWORKS_INFO[chainId] as EVMNetworkInfo)?.aggregator?.routerAddress
```
Comment (namgold) [inline src/hooks/useSwapV2Callback.ts:123]: use `networkInfo` from `useActiveWeb3React()`

File: src/hooks/useSwapV2Callback.ts [hunk 7]
```diff
@@ -162,5 +165,6 @@ export function useSwapV2Callback(
     walletSolana,
     addTransactionWithType,
     extractSwapData,
+    chainId,
```
Comment (namgold) [inline src/hooks/useSwapV2Callback.ts:168]:
```suggestion
    networkInfo.aggregator.routerAddress,
```

File: src/components/ApproveMessage/index.tsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+import { useActiveWeb3React } from 'hooks/index'
+import useTheme from 'hooks/useTheme'
+import { ExternalLink, TYPE } from 'theme/index'
+import { getEtherscanLink, shortenAddress } from 'utils/index'
+
+export default function ApproveMessage() {
+  const { chainId } = useActiveWeb3React()
+  const theme = useTheme()
+  const [showApproveMsgDetails, toggleShowApproveMsgDetails] = useToggle(true)
+  const EVMRouterAddress = (NETWORKS_INFO[chainId] as EVMNetworkInfo)?.aggregator?.routerAddress
```
Comment (namgold) [inline src/components/ApproveMessage/index.tsx:20]: why `?.` since they are not nullable

Comment (namgold) [reply src/components/ApproveMessage/index.tsx:20]: use networkInfo also

Comment (nguyenhuudungz) [reply src/components/ApproveMessage/index.tsx]: done

File: src/hooks/useSwapV2Callback.ts (hunk 7 above)
Comment (nguyenhuudungz) [reply src/hooks/useSwapV2Callback.ts:168]: done

**Review by nguyenhoaidanh — APPROVED** [review]

File: src/hooks/useApproveCallback.ts [hunk 9]
```diff
@@ -151,5 +151,6 @@ export function useApproveCallbackFromTradeV2(
     () => (trade ? computeSlippageAdjustedAmounts(trade, allowedSlippage)[Field.INPUT] : undefined),
     [trade, allowedSlippage],
   )
-  return useApproveCallback(amountToApprove, trade?.routerAddress || undefined)
+
+  return useApproveCallback(amountToApprove, trade?.routerAddress)
```
Comment (namgold) [inline src/hooks/useApproveCallback.ts:155]: this may misses case that `routerAddress` is empty string

**Pull request merged**